lib/awful/prompt.lua

```diff
@@ -518,10 +519,10 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                 hooks[key] = hooks[key] or {}
                 hooks[key][#hooks[key]+1] = v
             else
-                assert("The hook's 3rd parameter has to be a function.")
+                debug.print_warning("The hook's 3rd parameter has to be a function.")
             end
         else
-            assert("The hook has to have 3 parameters.")
+            debug.print_warning("The hook has to have 3 parameters.")
         end
```

Comment:
I'm fine with asserts too, I wonder how much of an issue it would be, passing wrong parameters simply doesn't work or doesn't working as intended, so maybe the impact would be minimal?
I'd go with @actionless 's solution.